ethcore/src/snapshot/service.rs

```diff
@@ -260,6 +297,37 @@ impl Service {
 		Ok(())
 	}
 
+	/// Take a snapshot at the block with the given number.
+	/// calling this while a restoration is in progress or vice versa
+	/// will lead to a race condition where the first one to finish will
+	/// have their produced snapshot overwritten.
+	pub fn take_snapshot(&self, client: &Client, num: u64) -> Result<(), Error> {
+		info!("Taking snapshot at #{}", num);
```

Comment:
Ok. How much of the progress do you think should be logged? As this is meant to be a pretty rare event I think it might be reasonable to print progress every 10 seconds or so?